src/Auth0.php

```diff
@@ -564,24 +564,20 @@ public function exchange()
      */
     public function renewTokens(array $options = [])
     {
-        if (! $this->accessToken) {
-            throw new CoreException('Can\'t renew the access token if there isn\'t one valid');
-        }
-
         if (! $this->refreshToken) {
             throw new CoreException('Can\'t renew the access token if there isn\'t a refresh token available');
         }
 
         $response = $this->authentication->refresh_token( $this->refreshToken, $options );
 
-        if (empty($response['access_token']) || empty($response['id_token'])) {
```

Comment:
Response could just be an access token so this check is not required.

Reply:
The response can also only contain an ID token if the user logged in with `response_type=id_token`. A Code Auth is the only one returning both artifacts as the final result unless you make a combination of them.

If your app will not be calling any APIs, it makes sense to only request and store an ID token.

Reply:
`response_type=id_token` would not generate an auth code and, hence, not end up in this flow. If you used `response_type=code` without an audience, you would also get a userinfo access token back. So an auth code exchange will always generate an access token.